### PR TITLE
Fixed view debug info by id

### PR DESCRIPTION
**Pull request by evil1**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #6

### CODE REVIEW

File: src/Repository/CollectorRepository.php [hunk 1]
```diff
@@ -63,7 +63,8 @@ private function loadData(): array
         $dataFiles = \glob($this->path . '/yii-debug*.data.json', GLOB_NOSORT);
         $data = [];
         foreach ($dataFiles as $file) {
-            $data = Json::decode(file_get_contents($file));
+            $id = \basename($file, '.data.json');
```
Comment (romkatsu) [inline src/Repository/CollectorRepository.php:66]: Duplicated with loadIndexData. I suggest we leave one method.

```php
private function loadData(string $fileSuffix): array
{
    clearstatcache();
    $dataFiles = \glob($this->path . '/yii-debug*' . $fileSuffix, GLOB_NOSORT);
    $data = [];
    foreach ($dataFiles as $file) {
        $id = \basename($file, $fileSuffix);
        $data[$id] = Json::decode(file_get_contents($file));
    }

    return $data;
}
```

```php
$this->loadData('.index.json');
$this->loadData('.data.json');
```

Comment (evil1) [reply src/Repository/CollectorRepository.php:66]: In this case we also need a filename template to be a second parameter for the method `getData()`

**Review by romkatsu — APPROVED** [review]

File: src/Repository/CollectorRepository.php [hunk 2]
```diff
@@ -30,40 +30,42 @@ public function getSummary(?string $id = null): array
...
-        foreach ($dataFiles as $file) {
-            $id = \basename($file, '.index.json');
-            $data[$id] = Json::decode(file_get_contents($file));
-        }
+        if (!empty($id)) {
+            $file = $this->path . DIRECTORY_SEPARATOR . $id . $fileSuffix;
+            if (file_exists($file) && !is_dir($file)) {
+                $id = \basename($file, $fileSuffix);
+                $data[$id] = Json::decode(file_get_contents($file));
+            }
```
Comment (rustamwin) [inline src/Repository/CollectorRepository.php:60]:
```suggestion
            if (!file_exists($file) && !is_dir($file)) {
                throw new NotFoundException(sprintf('Unable to find debug data ID with "%s"', $id));
            }
            
            return Json::decode(file_get_contents($file));
```

File: src/Repository/CollectorRepository.php [hunk 3]
```diff
@@ -30,40 +30,42 @@ public function getSummary(?string $id = null): array
...
 
+        if (empty($collector)) {
+            return $data[$id];
+        }
+
         if (!isset($data[$id][$collector])) {
             throw new NotFoundException(sprintf('Unable to find debug data collected with "%s"', $collector));
         }
 
         return $data[$id][$collector];
```
Comment (rustamwin) [inline src/Repository/CollectorRepository.php:48]:
```suggestion
        $data = $this->loadData('.data.json', $id);
        if (empty($collector)) {
            return $data;
        }

        if (!isset($data[$collector])) {
            throw new NotFoundException(sprintf('Unable to find debug data collected with "%s"', $collector));
        }

        return $data[$collector];
```

**Review by rustamwin — CHANGES_REQUESTED** [review]

**Review by xepozz — APPROVED** [review]

**Comment by rustamwin** [comment]
:+1:

**Pull request merged**